src/common/mfu_daos.c

```diff
@@ -2691,7 +2692,7 @@ static int serialize_recx_array(struct hdf5_args *hdf5,
             continue;
         for (i = 0; i < number; i++) {
             buf_len = recxs[i].rx_nr * size;
-            char buf[buf_len];
+            buf = MFU_CALLOC(1, buf_len);
```

Comment:
@daltonbohning I am always confused about this because is it not one element of buf_len size? I guess I see now it is buf_len number of elements the size of a char. But I think maybe in that case it would be MFU_CALLOC(recxs[I].rx_nr, size)?

Reply:
@dsikich I think it's still `MFU_CALLOC(buf_len, sizeof(char))`, because `recxs[I].rx_nr` is the number of `recxs`, and `size` is the size of EACH of those.
In total, we end up with
```
recxs[I].rx_nr * size * sizeof(char)
```
I think `MFU_CALLOC(recxs[I].rx_nr, size)` might technically work, but only because `sizeof(char) == 1`. If we wanted to do it this way, it might be better to use `void* buf` instead of `char* buf` for clarity.

Reply:
@daltonbohning this is the part why I am getting myself confused. I took it like number of elements is rx_nr without the size, and the second argument is the size of each element. I can update to using that way. I want to use it the correct way.

Reply:
@dsikich In most of the DAOS code base, buffers are void*, so I think using a void* buffer and `MFU_CALLOC(recxs[I].rx_nr, size)` is maybe okay

Reply:
@dsikich But then again, in DAOS, we do
```
D_ALLOC(buf, recxs[I].rx_nr * size)
```
Which I think translates to `calloc(recxs[I].rx_nr * size, 1)`

Reply:
@dsikich Ah, yes, D_ALLOC uses '1':
https://github.com/daos-stack/daos/blob/33f1724201d36a9c1807e7469c6f3bd9f75da2f7/src/include/gurt/common.h#L270